### PR TITLE
feat: capture V2 implementation (VF-2391)

### PR DESCRIPTION
**Pull request by DecathectZero**
**Fixes or implements VF-2391**

### Brief description. What is this change?

New capture block! We can just reuse the existing one for now for convenience.

### Related PRs

| branch              | PR          |
| ------------------- | ----------- |
| other_pr_production | [link](url) |
| other_pr_master     | [link](url) |

### Checklist

- [ ] this is a breaking change and should publish a new major version
- [ ] appropriate tests have been written

### CODE REVIEW

File: lib/services/dialog/index.ts [hunk 1]
```diff
@@ -35,38 +35,40 @@ export const utils = {
   isIntentInScope,
 };
 
-declare type DMStore = {
+export declare type DMStore = {
```
Comment (jonahsnider) [inline lib/services/dialog/index.ts:38]: not sure why this was a `declare type` to begin with, i would remove that
```suggestion
export type DMStore = {
```

File: lib/services/runtime/handlers/utils/entity.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  payload: { intent: { name }, query: '', entities: [] },
+});
+
+/** @description when VF_ELICIT is true, it will skip the entity prompt during entity filling */
+export const setElicit = (request: Request.IntentRequest, elicit: boolean): Request.IntentRequest & { [VF_ELICIT]: boolean } => ({
+  ...request,
+  [VF_ELICIT]: elicit,
+});
+
+export const hasElicit = (request: Request.IntentRequest & { [VF_ELICIT]?: boolean }): boolean => !!request[VF_ELICIT];
```
Comment (jonahsnider) [inline lib/services/runtime/handlers/utils/entity.ts:22]:
```suggestion
export const hasElicit = (request: Request.IntentRequest & { [VF_ELICIT]?: boolean }): request is Request.IntentRequest & { [VF_ELICIT]: true } =>
  request[VF_ELICIT] === true;
```

**Review by jonahsnider — APPROVED** [review]

File: lib/services/runtime/handlers/utils/entity.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export const EntityFillingNoMatchHandler = () => ({
+  handle: (node: NoMatchNode, runtime: Runtime, variables: Store) => (intents: string[] = [], defaultRequest?: Request.IntentRequest) => {
+    // see if the prior entity filling intent is within context
+    const priorIntent = runtime.storage.get<DMStore>(StorageType.DM)?.priorIntent;
+    const priorIntentMatch = intents.includes(priorIntent?.payload.intent.name!) && priorIntent?.payload.entities.length;
+
+    const nextRequest = (priorIntentMatch && setElicit(priorIntent!, false)) || defaultRequest;
+
+    // on first no match, explicitly prompt
+    // if (typeof runtime.storage.get(StorageType.NO_MATCHES_COUNTER) !== 'number' && nextRequest) {
```
Comment (trs) [inline lib/services/runtime/handlers/utils/entity.ts:35]: Any reason for keeping this commented code?

Comment (DecathectZero) [reply lib/services/runtime/handlers/utils/entity.ts:35]: don't think so

**Pull request merged**